src/main/groovy/de/dkfz/roddy/bam2fastq/ReadGroup.groovy

```diff
@@ -47,15 +47,15 @@ class ReadGroup {
     }
 
     /** Functional interface function doing a (shallow-)copy-on-write */
-    ReadGroup updatedFile(ReadFileType type, BaseFile newFile) {
-        Map<ReadFileType, BaseFile> newFiles = files.collectEntries { iterType, oldFile ->
+    ReadGroup<T> updatedFile(ReadFileType type, T newFile) {
```

Comment:
Here you still have the update method in place.

Reply:
This is generic it unconstrained T and thus a simple copys-on-write update method/copy construction.